include/optional.h

```diff
@@ -0,0 +1,100 @@
+// MIT License
+//
+// Created by iamOgunyinka (@iamOgunyinka, https://github.com/iamOgunyinka)
```

Comment:
@iamOgunyinka I tried to give proper attribution here, I hope it's ok with you

Reply:
it's OK haha it's your code, I don't need crediting.

Reply:
_Credit is where credit is due_